meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+asks whether it might be simpler to treat the LSP and BSP as orthogonal
+projects, and Stu and Jorge both agree.
+
+Jorge and Stu debate some of the interactions between LSP and BSP. Stu notes
+that the LSP server does not need a compiler. The discussion goes into some
+technical detail, without reaching a conclusion. Stu would like LSP servers to
+be thinner.
+
+Jon tries with limited success to keep the conversation focussed away from
+technical details.
+
+Eugene Burmako added that he sees SemanticDB as a useful standard that works
+for both complete and partial information. He points out that it doesn't have
+to be persisted, so it can be exchanged in-memory. He notes that it's used to
+index third-party dependencies. It is used to communicate with the presentation
+compiler: the presentation compiler can produce SemanticDB data. Eugene
+suggests that SemanticDB could be a good standard for exchanging information
+between tools.
+
+Jon suggests that the LSP proposal is well-defined, and implementations exist,
+but suggests that the BSP should be designed to be modular to support more
+heterogeneous features. He suggest that different people could be involved with
+working on the two proposals.
+
+Shane Delmore asks about timescales for development, and asks what we're trying
+to support, and mentioned that people may want to work on projects which
+interoperate with other languages. Shane asks whether TASTY works currently
+with Scalac 2.11 or 2.12, and Adriaan says there's currently no schedule for
+including TASTY support in Scalac 2.13. He's open to it, but thinks it's more
+of a Scalac 2.14 target.
+
+Jon claims (in Martin's absence) that Martin didn't think it would be a huge
+amount of work to get a basic implementation of TASTY working for Scalac though
+challenges would exist in deciding the exact representations of Scalac
+structures in a format designed primarily with Dotty in mind, and Adriaan
+thinks that it's a challenge to get TASTY to scale between Java, Scala and
+Dotty.
+
+Jorge suggests some solutions for cross-language support.
+
+## Next Steps
+
+With limited time remaining in the meeting, Jon makes a proposal that two
+working groups should be formed to work on the LSP and the BSP, made up of
+different people with some overlap between the two, and that the working groups
+should come under the remit of the Scala Center.
+
+Jon notes that the work of the Scala Center should be recommended by the
+Advisory Board, and that it would be good to have a proposal from one of the
+sponsors so that an official vote could be taken during a Scala Center Advisory
+Board meeting, proposing that the Scala Center support the initiative.
+
+Jon suggests six to eight people should be in each group.
+
+Stu suggests a "protocols subgroup" within the Scala Center, and suggests that
+Twitter could put in a proposal for the next meeting.
+
+Jorge is skeptical about what can be achieved without having a good definition
```

Comment:
I meant not what the groups can do, but what problem we’re trying to solve. During the talk I had some confusion to distinguish whether we were talking about lsp, bsp or a mix of the two.

My point was we cannot know if we need a protocol subgroup because there are many moving variables. I think it would be beneficial to have another meeting before we decide to create such groups. Let’s first agree on the problems, I got a feeling we were not on the same page there...

Reply:
I think that it would be more work than necessary to have another large meeting to decide on this, and would just be a further delay. I would suggest instead that we form a "protocols" group as Stu suggested. If we run the group, *in some form* for a month prior to the next Scala Center Advisory Board meeting in 5-6 weeks, Stu can put in a proposal which takes this experience into account.

More generally, I don't think there's any big problem with modifying (or disbanding) the nature working group(s) after they've formed.

I'll change the wording a bit here. Take a look again once I update the PR and let me know if you need me to change it further.